Java/core/src/main/java/com/amazon/randomcutforest/store/SmallLeafStore.java

```diff
@@ -90,6 +90,12 @@ public int getPointIndex(int index) {
         return pointIndex[index];
     }
 
+    @Override
+    public int setPointIndex(int index, int pointIndex) {
+        this.pointIndex[index] = pointIndex;
+        return mass[index];
```

Comment:
Is this method really supposed to return mass?

Reply:
It's a useful shortcut which avoids asking for the mass a second time. But not really needed.

Reply:
It's actually helpful to return the older stored index (to audit).